### ISSUE TITLE
Lines randomly disappear when switching between panes

### ISSUE THREAD

**Issue by Don-Vito** [microsoft/terminal#9203]
# Environment
Main development branch

# Steps to reproduce

1. Enable FFM mode (not sure if required)
2. Open a Windows Powershell tab with two panes
3. In repro scenario, the panes contain a link (not sure if required)
3. Move the mouse from one pane to another

# Expected behavior

The panes get focus, the links get underlined / de-underlined.

# Actual behavior
Randomly some lines disappear from a pane and reappear after some hovering (see below)

![DisappearingLine](https://user-images.githubusercontent.com/4639110/108351983-e6025600-71ee-11eb-8a84-02b338e179a4.gif)

My guess it is somehow related to us triggering redraw upon pointer move / exit.

**Comment by zadjii-msft** [comment]
what the _heck_...?

Which commit are you on exactly? That's mostly just for reference to see if anything specific got committed recently. I can't get it to repro on Version: 1.7.413.0 _without FFM_, but I'm not sure that build even has FFM. 


Dang `Renderer.cpp:185` is just the generic catch-all for the render frame

**Comment by Don-Vito** [comment]
I am using the current main. This can be related to me adding De-underlining hyperlinks yesterday, that might trigger extra redraw.

**Comment by zadjii-msft** [comment]
I'm gonna /cc @miniksa since he's always interested in weird renderer error messages like that. 

@Don-Vito you mind if I assign you since you've got the live repro (and are literally updating this post as I reply 😛)?

**Comment by Don-Vito** [comment]
> 
> 
> I'm gonna /cc @miniksa since he's always interested in weird renderer error messages like that.
> 
> @Don-Vito you mind if I assign you since you've got the live repro (and are literally updating this post as I reply 😛)?

Opened is a ticket as I have no clue in rendering :blush: I will be glad to continue researching it.. but I guess it might be not that efficient :smile:

**Comment by zadjii-msft** [comment]
No worries then - I'll try and loop back on this before 1.7 closes out

**Comment by miniksa** [comment]
Sorry, I don't have an idea here off the top of my head. But I am curious/interested in what's going on. I didn't put in either the hyperlink drawing nor the swap chain waitable object thing, so I have less of an idea than I typically do.

If I get time, I'll come back around to this, but I'm hoping someone else can dig.

**Comment by Don-Vito** [comment]
When you see the comments like this:
```
const auto ret = WaitForSingleObjectEx(
        _swapChainFrameLatencyWaitableObject.get(),
        1000, // 1 second timeout (shouldn't ever occur)
        true);
```
you know these are the famous last words :smile:
I am getting timeout, mostly in debug mode with debugger attached.

**Comment by miniksa** [comment]
> When you see the comments like this:
> 
> ```
> const auto ret = WaitForSingleObjectEx(
>         _swapChainFrameLatencyWaitableObject.get(),
>         1000, // 1 second timeout (shouldn't ever occur)
>         true);
> ```
> 
> you know these are the famous last words 😄
> I am getting timeout, mostly in debug mode with debugger attached.

640KB all over again. Oh no. Maybe it just needs a debugging escape hatch?

**Comment by PankajBhojwani** [comment]
I can repro the error messages in the output from just repeatedly hovering and exiting a link (no second pane required) - not sure if this was there from when links were introduced or due to the more recent hover underline changes.

Can't seem to get the line disappearing to repro though

**Comment by Don-Vito** [comment]
And the line with the comment is taken directly from the official MSFT documentation :smile:
https://docs.microsoft.com/en-us/windows/uwp/gaming/reduce-latency-with-dxgi-1-3-swap-chains

![image](https://user-images.githubusercontent.com/4639110/108426268-3e177780-7244-11eb-8063-030c298c3b1c.png)

**Comment by Don-Vito** [comment]
> 
> 
> I can repro the error messages in the output from just repeatedly hovering and exiting a link (no second pane required) - not sure if this was there from when links were introduced or due to the more recent hover underline changes.
> 
> Can't seem to get the line disappearing to repro though

Yes. Actually I am not even sure if the message is relevant for the line disappearing. There is some correlation, but from the code I don't see that rendering is stopped.
